core/nuxeo-page-provider.js

```diff
@@ -428,26 +428,26 @@ import './nuxeo-resource.js';
         params.query = this.query;
       }
 
-      if (this.params.queryParams) {
-        // these should've simply been passed as an array instead
-        params.queryParams = this.params.queryParams;
-      } else if (Array.isArray(this.params)) {
-        params.queryParams = this.params;
-      } else {
-        const namedParams = {};
-        Object.keys(this.params).forEach((key) => {
-          let value = this.params[key];
-          if (value != null) {
-            if (Array.isArray(value)) {
-              value = JSON.stringify(value.map((item) => (item['entity-type'] ? item.uid || item.id : item)));
-            } else if (typeof value !== 'string') {
-              value = value['entity-type'] ? value.uid || value.id : JSON.stringify(value);
-            }
-            namedParams[key] = value;
+      const namedParams = {};
+      Object.keys(this.params).forEach((key) => {
+        if (key === 'queryParams') {
+          const queryParams = this.params[key];
+          if (queryParams) {
+            params.queryParams = queryParams;
           }
-        }, this);
-        params.namedParameters = namedParams;
-      }
+          return;
+        }
+        let value = this.params[key];
+        if (value != null) {
```

Comment:
i suggest either `if(value)` or `if (value !== null)`

Reply:
also to avoid complexity, maybe if(!value) return

Reply:
Please note that what you suggested is no equivalent to what we have here. `value != null` is used to check for both not `undefined` or not `null`, whereas just checking for `value` would also also check for no empty strings.